tests/collections/collection.test.ts

```diff
@@ -80,9 +80,74 @@ describe(`StargateMongoose - ${testClientName} Connection - collections.collecti
     it('Should fail insert of doc over size 1 MB', async () => {
       const jsonDocGt1MB = new Array(1024 * 1024).fill("a").join("");
       const docToInsert = { username: jsonDocGt1MB };
-      collection.insertOne(docToInsert).catch((e) => {
-        assert.strictEqual(e.errors[0].message, "Request invalid, the field postCommand.command.documents not valid: document size is over the max limit.");
-      });
+      let error: any;
+      try {
+        await collection.insertOne(docToInsert);
+      } catch (e: any) {
+        error = e;
+      };
+      assert.ok(error);
+      assert.strictEqual(error.errors[0].message, "Request failed with status code 413");
+    });
+    it('Should fail if the number of levels in the doc is > 8', async () => {
+      const docToInsert = { l1: { l2: { l3: { l4: { l5: { l6: { l7: { l8: { l9: "l9value" } } } } } } } } };
+      let error: any;
+      try {
+        await collection.insertOne(docToInsert);
+      } catch (e: any) {
+        error = e;
+      };
+      assert.ok(error);
+      assert.strictEqual(error.errors[0].message, "Document size limitation violated: document depth exceeds maximum allowed (8)");
+    });
+    it('Should fail if the field length is > 48', async () => {
+      const fieldName = 'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvw';
```

Comment:
Strings have a nice `repeat()` function, so probably cleaner to just do `'a'.repeat(49)` to generate a length 49 string.